README.md

```diff
@@ -1,3 +1,141 @@
+
+We now apply our dogmatism classifier to a larger dataset of posts, examining how dogmatic language shapes the Reddit community. Concretely, we apply the BOW+LING model trained on the full Reddit dataset to millions of new unannotated posts, labeling these posts with a probability of dogmatism according to the classifier (0=non-dogmatic, 1=dogmatic). We then use these dogmatism annotations to address four research questions."""
+
+snippet = "Concretely, we apply the BOW+LING model trained on the full Reddit dataset to millions of new unannotated posts, labeling these posts with a probability of dogmatism according to the classifier (0=non-dogmatic, 1=dogmatic)."
+
+decontextualize(snippet, context)
+> "[REF0] apply the BOW+LING [bag-of-words and linguistic features] model trained on the full Reddit dataset [different subreddit representing different topics, such as politics, business, science and other other posts in the Reddit home page] to millions of new unannotated posts, labeling these posts with a probability of dogmatism according to the classifier (0=non-dogmatic, 1=dogmatic)."
+```
+
+Often times, you want to use a whole paper or a list of papers as context. You can specify the papers as paths to `json` files with a specific format or as S2ORC IDs.
+```python
+paper_1 = "path/to/paper.json"
+paper_2 = "path/to/cited_paper.json"
+
+decontextualize(snippet, paper_1)
+decontextualize(snippet, [paper_1, paper_2])
+```
+
+Below is the `json` file format:
+```json
+{
+    "title": "<title>",
+    "abstract": "<abstract>",
+    "full_text": [{
+        "section_name" : "<section_title>",
+        "paragraphs": ["<paragraph>", ...]
+    }, ...]
+}
+```
+Subsection names should be separated from their supersection name with ":::". For example, the subsection "Metrics" of the "Methods" section would have the `section_name: "Methods ::: Metrics"`.
+
+You can also specify papers with S2ORC IDs if the full text of the paper as long as the full text of the paper is in S2ORC.
+```python
+decontextualize(snippet, "52118895")
+decontextualize(snippet, ["52118895", "65318895"])
+```
+
+3. Customize decontextualization pipeline
+
+By default we use GPT4 to answer based on the full document, but you can customize the different part of the pipeline by including a config in the call to `decontextualize`. The pipeline consists of three parts: A question generator (`qgen`), a question-answering system (`qa`), and a synthesizer (`synth`) to rewrite the snippets to include the answers to the questions. Each component of the pipeline can be specified using the a config. The config can either be a dictionary or a path to a yaml file with the following structure. (The values are the defaults.)
+```yaml
```

Comment:
Have a Pydantic for the config as well.
Allow for passing openai key here as well.